src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const tbody = document.querySelector('tbody');
+const thead = document.querySelector('thead');
+const rowArray = [...tbody.rows];
+
+const getSellValue = (tr, i) => {
+  return tr.children[i].innerText;
+};
+
+thead.addEventListener('click', (e) => {
+  const th = e.target;
+  const idx = th.cellIndex;
+
+  rowArray.sort((rowA, rowB) => {
+    const cellA = getSellValue(rowA, idx);
+    const cellB = getSellValue(rowB, idx);
+
+    const a = isNaN(cellA) ? cellA : parseFloat(cellA.replace(/[^0-9.]/g, ''));
+
+    const b = isNaN(cellB) ? cellB : parseFloat(cellB.replace(/[^0-9.]/g, ''));
```

Comment:
The code attempts to parse cell values as numbers if possible, which is a good approach for sorting. However, if the column contains mixed data types (e.g., numbers and strings), this logic might not work as expected. Consider adding a check to ensure that the column data type is consistent or handle mixed types more explicitly.